### PR TITLE
Add `skipHandlers` method

### PR DESCRIPTION
**Pull request by SentryMan**
Sometimes you need to skip other handlers

### CODE REVIEW

File: avaje-sigma/src/main/java/io/avaje/sigma/core/SigmaContext.java [hunk 1]
```diff
@@ -232,4 +238,8 @@ public AWSHttpResponse createResponse() {
     return new AWSHttpResponse(
         status, responseHeaders, multiValueResponseHeaders, body, base64Encoded);
   }
+
+  public boolean handlersSkipped() {
```
Comment (rbygrave) [inline avaje-sigma/src/main/java/io/avaje/sigma/core/SigmaContext.java:242]: Looks like this should be part of the HttpContext API.

File: avaje-sigma/src/main/java/io/avaje/sigma/routes/SpiRoutes.java [hunk 2]
```diff
@@ -11,10 +13,10 @@ public interface SpiRoutes {
   Entry match(Router.HttpMethod type, String pathInfo);
 
   /** Execute all appropriate before filters for the given request URI. */
-  void before(String pathInfo, HttpContext ctx) throws Exception;
+  void before(String pathInfo, SigmaContext ctx) throws Exception;
```
Comment (rbygrave) [inline avaje-sigma/src/main/java/io/avaje/sigma/routes/SpiRoutes.java:16]: Why use the concrete SigmaContext here in the API? I'm thinking that `handlersSkipped()` probably should be public API of HttpContext.

That is, for me I'm aiming for an API to be interfaces (purely abstract). It looks like HttpContext should still work if `handlersSkipped()` was made part of the API.

Once we use concrete classes like SigmaContext in the API it becomes harder for folks to use test doubles etc and restricts the ability to change the implementation over time so trying to keep the API as interfaces is imo good practice.

File: avaje-sigma/src/main/java/io/avaje/sigma/core/SigmaContext.java (hunk 1 above)
Comment (SentryMan) [reply avaje-sigma/src/main/java/io/avaje/sigma/core/SigmaContext.java]: don't care either way, but for what purpose? Any handler that calls  `skipRemainingHandlers` is the one skipping, what use has it for a method that tells you if you have already skipped?

It's not like the skipped handlers have any chance to use it either, it's purely for the internals.

File: avaje-sigma/src/main/java/io/avaje/sigma/routes/SpiRoutes.java (hunk 2 above)
Comment (SentryMan) [reply avaje-sigma/src/main/java/io/avaje/sigma/routes/SpiRoutes.java]: > Once we use concrete classes like SigmaContext in the API it becomes harder for folks to use test doubles etc and restricts the ability to change the implementation over time so trying to keep the API as interfaces is imo good practice.

Possibly, but SPIRoutes is an internal API not meant to be used externally

**Comment by SentryMan** [comment]
@rbygrave any more thoughts on this?

**Review by rbygrave — APPROVED** [review]

**Pull request merged**